unidist/core/backends/mpi/core/communication.py

```diff
@@ -116,39 +164,74 @@ def get_instance(cls, *args):
             cls.__instance = MPIState(*args)
         return cls.__instance
 
+    def is_root_process(self, rank=None):
+        """
+        Check if the rank is root process.
 
-class MPIRank:
-    """Class that describes ranks assignment."""
+        Parameters
+        ----------
+        rank : int, optional
+            The rank to be checked.
 
-    ROOT = 0
-    MONITOR = 1
-    FIRST_WORKER = 2
+        Returns
+        -------
+        bool
+            True or False.
+        """
+        if rank is None:
+            rank = self.global_rank
+        return rank == MPIRank.ROOT
 
+    def is_monitor_process(self, rank=None):
+        """
+        Check if the rank is a monitor process.
 
-def get_topology():
-    """
-    Get topology of MPI cluster.
+        Parameters
+        ----------
+        rank : int, optional
+            The rank to be checked.
 
-    Returns
-    -------
-    dict
-        Dictionary, containing workers ranks assignments by IP-addresses in
-        the form: `{"node_ip0": [rank_2, rank_3, ...], "node_ip1": [rank_i, ...], ...}`.
-    """
-    mpi_state = MPIState.get_instance()
-    comm = mpi_state.comm
-    rank = mpi_state.rank
+        Returns
+        -------
+        bool
+            True or False.
+        """
+        if rank is None:
+            rank = self.global_rank
+        return rank in self.monitor_processes
+
+    def get_monitor_by_worker_rank(self, global_rank=None):
```

Comment:
```suggestion
    def get_monitor_by_worker_rank(self, rank=None):
```
In the methods `is_root_process` and `is_monitor_process` you use `rank` as a parameter but here `global_rank`. Let's align the parameters for the methods, e.g., `global_rank` or `rank`.

Reply:
left rank